var/spack/repos/builtin/packages/cabana/package.py

```diff
@@ -59,5 +59,19 @@ def cmake_args(self):
             '-DBUILD_SHARED_LIBS=%s' % (
                 'On' if '+shared'  in self.spec else 'Off')
         ]
+        # These variables were removed in 0.3.0 (where backends are
+        # automatically used from Kokkos)
+        if self.spec.satisfies('@:0.2.0'):
+            backends = {'serial': 'Serial',
+                        'openmp': 'OpenMP',
+                        'cuda': 'Cuda'}
+            for backend in backends:
+                options.append('-DCabana_ENABLE_%s=%s' %
+                               (backends[backend],
+                                'On' if '+' + backend in self.spec else 'Off'))
```

Comment:
The `CMakePackage.define_from_variant` will help here (and check to make sure the variant names are valid):
```suggestion
                options.append(self.define_from_variant('Cabana_ENABLE_' + backends[backend], backend))
```